### PR TITLE
Construct: Better 404 Pages

### PR DESCRIPTION
**Pull request by gdjohnson**
This branch replaces the previous 404 page with a Construct EmptyState page, and fixes an infinite spinner on user profiles.

## Description
The PageLoading component now wraps a Construct EmptyState component with an x-octagon icon; it can take a custom message but provides generic defaults.

Elsewhere, on address profile pages, if the address was invalid, a loading spinner would render indefinitely. This branch implements a simple count to track attempts at validating the address, and quickly switches from a spinner to a 404 page warning the visiting user that the address appears invalid.

image

## How has this been tested?
Corrupted & valid addresses were tested for proper behavior. The 404 pages were tested logged-in and logged-out. Styling was tested on different browsers and window sizes from size md to xl (xs and sm, i.e. mobile styling, will require larger structural changes to sidebar & switcher, and was disregarded).

## Have proper tags been added (for bug, enhxs and smancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [ ] yes, and they are tested: [enter the % coverage here]
- [ ] yes, and they are not tested: [enter the % coverage here]
- [ ] yes, but I did not run tests
- [x] no

### CODE REVIEW

File: client/scripts/views/pages/profile/index.ts [hunk 1]
```diff
@@ -188,14 +189,23 @@ export enum UserContent {
...
+const ProfilePage: m.Component<{ address: string }, { loadCount: number }> = {
   oncreate: (vnode) => {
     mixpanel.track('PageVisit', { 'Page Name': 'LoginPage' });
   },
   view: (vnode) => {
+    if (!vnode.state.loadCount) vnode.state.loadCount = 0;
     if (!app.chain) return m(PageLoading);
     const account = app.chain.accounts.get(vnode.attrs.address);
-    if (!account) return m(PageLoading);
+    console.log(vnode.state.loadCount);
```
Comment (drewstone) [inline client/scripts/views/pages/profile/index.ts:200]: I'd remove logs

Comment (gdjohnson) [reply client/scripts/views/pages/profile/index.ts:200]: ooh thank ya!

**Comment by dillchen** [comment]
This looks good to me, checked out and repeated the logged in and logged out tests on my side. Think this could be merged, probably doesn't need another review from @raykyri, imo.

**Comment by raykyri** [comment]
The logic around loadCount should have at least an explanation for why it is that way. Can you explain the rationale for that check? It feels brittle and seems like it could cause unwanted interactions, e.g. when if we ever move from one profile page to another.

**Comment by raykyri** [comment]
So to double back here, I would like to figure out why we need the logic for loadCount. However this is lower priority than the other issues we have filed for this week so we can also put this on hold for a little.

**Comment by gdjohnson** [comment]
Yep, this is entirely my bad. I instituted it based on a concern that I did not test, and on testing, the concern turned out not to be justified. loadCount code has been removed.

**Pull request merged**